oauth2_client/client.py

```diff
@@ -45,7 +88,21 @@ def request(self, method, url, *args, **kwargs):  # pylint: disable=arguments-di
             Response object.
         """
         absolute_url = urljoin(self.service_host, url)
-        return super(OAuth2Client, self).request(method, absolute_url, *args, **kwargs)
+        try:
+            resp = super(OAuth2Client, self).request(method, absolute_url, *args, **kwargs)
+            if resp.status_code == 403:
+                raise TokenExpiredError()
+            else:
+                return resp
+        except TokenExpiredError:
+            log.warning("Access Token for app=%s expired.", self.app.name)
+            if self.expired_callback:
+                new_token = self.expired_callback(self.app)
+                self.token = new_token.to_client_dict()
+                log.info("Obtained new token. Retrying request.")
+                self.request(method, url, *args, **kwargs)
```

Comment:
btw, shouldn't `return` be used in this line?

Reply:
hell yes, it should! thank you